### PR TITLE
Add API for frameworks and examples

### PR DESCRIPTION
**Pull request by AndyBitz**
This adds a Next.js page to this repo with an API for the examples and the frameworks. This makes sure that the frameworks API always provides the latest data when it gets updated.

The API is not used for now, after it gets merged I'll add `examples/` and when this is done we'll replace the existing `frameworks` and `examples` API with this one.

[PRODUCT-944]
[PRODUCT-955]

[PRODUCT-944]: https://zeit.atlassian.net/browse/PRODUCT-944
[PRODUCT-955]: https://zeit.atlassian.net/browse/PRODUCT-955

### CODE REVIEW

**Comment by vercel-team** [comment]
#### 📝Changed routes:



- [**/api/examples/download/\[segment\]**](https://now-git-frameworks-update.zeit.now.sh/api/examples/download/[segment])
- [**/api/examples/info**](https://now-git-frameworks-update.zeit.now.sh/api/examples/info)
- [**/api/examples/list-all**](https://now-git-frameworks-update.zeit.now.sh/api/examples/list-all)
- [**/api/examples/list**](https://now-git-frameworks-update.zeit.now.sh/api/examples/list)
- [**/api/frameworks**](https://now-git-frameworks-update.zeit.now.sh/api/frameworks)



Commit 7dbab1ac1e7d35e76a874d2b5b660593ca69015f (https://now-git-frameworks-update.zeit.now.sh).

File: package.json [hunk 1]
```diff
@@ -12,20 +12,32 @@
...
     "buffer-replace": "1.0.0",
     "eslint": "6.2.2",
     "eslint-config-prettier": "6.1.0",
     "husky": "3.0.4",
     "lint-staged": "9.2.5",
     "node-fetch": "2.6.0",
-    "prettier": "1.18.2"
+    "prettier": "1.18.2",
+    "typescript": "3.7.4",
+    "yarn": "1.21.1"
```
Comment (styfle) [inline package.json]: Why is `yarn` here?

File: now.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "redirects": [
+    {
+      "source": "/",
+      "destination": "https://zeit.co/new"
```
Comment (styfle) [inline now.json]: Should we make this a temporary redirect (`statusCode: 307`)?

File: package.json [hunk 3]
```diff
@@ -12,20 +12,32 @@
...
+    "tar-fs": "2.0.0",
+    "unzip-stream": "0.3.0"
   },
   "devDependencies": {
+    "@types/node": "13.1.4",
+    "@types/react": "16.9.17",
     "@typescript-eslint/eslint-plugin": "2.0.0",
     "@typescript-eslint/parser": "2.0.0",
     "@zeit/ncc": "0.20.4",
+    "add": "2.0.6",
```
Comment (styfle) [inline package.json]: What is `add`? I dont see it used.

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "compilerOptions": {
+    "target": "es5",
```
Comment (styfle) [inline tsconfig.json]:
```suggestion
    "target": "esnext",
```

Comment (styfle) [reply tsconfig.json]: I think you can avoid transpiling and let Next.js add polyfills right?

File: package.json [hunk 5]
```diff
@@ -12,9 +12,18 @@
...
   "dependencies": {
-    "lerna": "3.16.4"
+    "got": "10.2.1",
+    "lerna": "3.16.4",
+    "next": "9.1.7",
+    "parse-github-url": "1.0.2",
+    "react": "16.12.0",
+    "react-dom": "16.12.0",
+    "tar-fs": "2.0.0",
+    "unzip-stream": "0.3.0"
```
Comment (styfle) [inline package.json]: Can you revert these changes and instead create `/api/package.json` with the necessary deps?

Comment (AndyBitz) [reply package.json]: @styfle we need it in the root in order to make Next.js work though, don't we

Comment (styfle) [reply package.json]: @AndyBitz I don't think so. Furthermore, why are we using Next.js if this is only an API? Are we expecting to build a frontend in the future?

Comment (AndyBitz) [reply package.json]: @styfle likely, also, we need to build the root, because we need `packages/frameworks` as well, if we'd put it inside a subdirectory we wouldn't be able to.

Comment (styfle) [reply package.json]: Why not? I believe that node-file-trace will read `packages/frameworks` properly if you import `@now/frameworks` from the api dir.

Comment (AndyBitz) [reply package.json]: I had to use `../../packages/frameworks` in order to make it work, probably because `api` is not a package for `lerna` (not sure if it should be)?

Comment (styfle) [reply package.json]: I think `../../packages/frameworks` is acceptable, as long as TS can figure it out 👍

File: api/frameworks.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { NowRequest, NowResponse } from '@now/node';
+import { withApiHandler } from './_lib/util/with-api-handler';
+import frameworkList, { Framework } from '../packages/frameworks';
+
+const frameworks: Framework[] = (frameworkList as Framework[]).map(
```
Comment (styfle) [inline api/frameworks.ts:5]: Hopefully we can avoid `as` in a future PR when `@now/frameworks` is updated 😄

File: api/package.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "api",
+  "private": true,
+  "version": "0.0.0",
+  "description": "API for the zeit/now repo",
+  "main": "index.js",
+  "scripts": {
+    "build": "tsc",
+    "now-build": "yarn --cwd .. && node ../run.js build all && tsc"
```
Comment (styfle) [inline api/package.json]: I'm not sure if you need `tsc` here since that will already happen for `@now/node`

File: api/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,15 @@
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "skipLibCheck": true,
+    "strict": false,
+    "forceConsistentCasingInFileNames": true,
+    "noEmit": true,
+    "esModuleInterop": true,
+    "module": "esnext",
```
Comment (styfle) [inline api/tsconfig.json]:
```suggestion
    "module": "commonjs",
```
I'm pretty sure you can't use ESM with Node 12. If we aren't using Next.js, then we'll need to convert to CJS.

**Review by styfle — APPROVED** [review]
Looks good, nice job! 🎉

File: api/frameworks.ts (hunk 6 above)
Comment (AndyBitz) [reply api/frameworks.ts:5]: Yup :D

File: api/package.json (hunk 7 above)
Comment (AndyBitz) [reply api/package.json]: yes, looks like it works without it 👍
